geomesa-fs/geomesa-fs-datastore/src/main/scala/org/locationtech/geomesa/fs/storage/converter/ConverterStorage.scala

```diff
@@ -0,0 +1,121 @@
+import org.locationtech.geomesa.fs.storage.api._
+import org.locationtech.geomesa.fs.storage.common.{StoragePartition, PartitionScheme}
+import org.locationtech.geomesa.utils.geotools.{SftArgResolver, SftArgs}
+import org.opengis.feature.simple.SimpleFeatureType
+
+
+class ConverterStorageFactory extends FileSystemStorageFactory {
+  override def canProcess(params: util.Map[String, Serializable]): Boolean = {
+    params.containsKey("fs.path") &&
+    params.containsKey("fs.encoding") &&
+      params.get("fs.encoding").asInstanceOf[String].equals("converter")
+  }
+
+  override def build(params: util.Map[String, Serializable]): FileSystemStorage = {
+    val path = params.get("fs.path").asInstanceOf[String]
+    val root = new Path(path)
+
+    val sftArg = Option(FileSystemDataStoreParams.SftConfigParam.lookUp(params))
+      .orElse(Option(FileSystemDataStoreParams.SftNameParam.lookUp(params)))
+      .map(_.asInstanceOf[String])
+      .getOrElse( throw new IllegalArgumentException(s"Must provide sft config or name"))
+
+    val sft = SftArgResolver.getArg(SftArgs(sftArg, null)) match {
+      case Left(e) => throw e
+      case Right(sftype) => sftype
+    }
+
+    val convertArg = Option(FileSystemDataStoreParams.ConverterConfigParam.lookUp(params)).
+      orElse(Option(FileSystemDataStoreParams.ConverterNameParam.lookUp(params)))
+      .map(_.asInstanceOf[String])
+      .getOrElse(throw new IllegalArgumentException(s"Must provide either converter config or name"))
+
+    val converterConfig = ConverterConfigResolver.getArg(ConfArgs(convertArg)) match {
+      case Left(e) => throw e
+      case Right(conf) => conf
+    }
+    val converter = SimpleFeatureConverters.build(sft, converterConfig)
+
+    val partitionScheme = PartitionScheme(sft, params)
+    new ConverterStorage(root, root.getFileSystem(new Configuration), partitionScheme, sft, converter)
+  }
+}
+
+
+class ConverterStorage(root: Path,
+                       fs: FileSystem,
+                       partitionScheme: PartitionScheme,
+                       sft: SimpleFeatureType,
+                       converter: SimpleFeatureConverter[_]) extends FileSystemStorage {
+  import scala.collection.JavaConversions._
+
+  override def listTypeNames(): util.List[String] = List(sft.getTypeName)
+
+  override def listFeatureTypes(): util.List[SimpleFeatureType] = List(sft)
+
+  override def createNewFeatureType(sft: SimpleFeatureType, partitionScheme: PartitionScheme): Unit =
+    throw new UnsupportedOperationException("Cannot create new feature type on existing DB")
+
+  override def getFeatureType(name: String): SimpleFeatureType =
+    if (sft.getTypeName != name) {
+      throw new IllegalArgumentException(s"Type $name doesn't match configured sft name ${sft.getTypeName}")
+    } else {
+      sft
+    }
+
+  override def getWriter(typeName: String, partition: Partition): FileSystemWriter =
+    throw new UnsupportedOperationException("Cannot append to converter datastore")
```

Comment:
"Cannot append to converter datastore."

Also, I'd make this consistent with the above message.